server/tests/host_test.cpp

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <vector>
+#include <numeric>
+#include <map>
+#include <string>
+
+#include "../host/host.cpp"
+
+#include "../../common/encryption/encrypt.h"
+#include "../../common/encryption/serialization.h"
+
+using namespace std;
+
+int main(int argc, char* argv[]) 
+{  
+    size_t accumulator_length = 3;
+    unsigned char*** encrypted_accumulator = new unsigned char**[accumulator_length * sizeof(unsigned char**)];
+    size_t* accumulator_lengths = new size_t[accumulator_length * sizeof(size_t)];
+
+    for (int i = 0; i < accumulator_length; i++) {
+        map<string, vector<double>> accumulator = {{"w1", {i, i + 1, i + 2, i + 3}}, 
+                                                    {"w2", {i + 1, i + 2, i + 3, i + 4}}};
+        string accumulator_s = serialize(accumulator);
+
+        *(encrypted_accumulator + i) = new unsigned char*[3 * sizeof(unsigned char*)];
+        encrypt_bytes((unsigned char*) accumulator_s.c_str(), accumulator_s.size(), *(encrypted_accumulator + i));
+        *(accumulator_lengths + i) = accumulator_s.size();
```

Comment:
I think you should be able to assign by just doing `accumulator_lengths[i]` instead of `*(accumulator_lengths + i)`